backend/src/db/repositories/CommentRepository.ts

```diff
@@ -43,23 +43,36 @@ export default class CommentRepository {
             });
     }
 
-    async getUserComments(userId: number, forUserId: number, page: number, perPage: number): Promise<CommentRawWithUserData[]> {
+    async getUserComments(userId: number, forUserId: number, filter: string, page: number, perPage: number): Promise<CommentRawWithUserData[]> {
         const limitFrom = (page - 1) * perPage;
-        return await this.db.query(`
+        let query = `
                 select c.*, v.vote
                     from comments c
                       left join comment_votes v on (v.comment_id = c.comment_id and v.voter_id = :for_user_id)
                     where
                       c.author_id = :user_id and c.deleted = 0
                     order by c.created_at desc
                     limit :limit_from, :limit_count
-            `,
-            {
-                user_id: userId,
-                for_user_id: forUserId,
-                limit_from: limitFrom,
-                limit_count: perPage
-            });
+            `;
+          const params = {
+              user_id: userId,
+              for_user_id: forUserId,
+              limit_from: limitFrom,
+              limit_count: perPage
+          };
+        if (filter) {
+            query = `
+                select c.*, v.vote
+                    from comments c
+                      left join comment_votes v on (v.comment_id = c.comment_id and v.voter_id = :for_user_id)
+                    where
+                      c.author_id = :user_id and c.deleted = 0 and c.source like :filter
```

Comment:
Let's maybe use full text index here?

Reply:
I am unsure whether we should use it here and I am debating with myself, but I tend to lean towards not using it. This would duplicate advanced search functionality once it's there

Reply:
Ok, if that's just a temporary solution to be replaced with elasticsearch, I'm ok with it.